### DISCUSSION TITLE
Calculator with Reactive ui

### DISCUSSION THREAD

**Discussion by mGKScoder** [reactiveui/ReactiveUI#2937] [Q&A]
Hello Everyone. I have to create a calculator with WPF MVVM + Reactiveui. Can someone help me about that ? I have to complete this task in this week.

**Comment by ChrisPulman** [comment]
Which part are you stuck with? 
Did you read the documentation?
https://www.reactiveui.net/docs/handbook/

**Reply by mGKScoder** [reply]
I finished the viewmodel part. Now I need to create a model somehow and bind it with a view. But I have no idea how to do it.

**Comment by glennawatson** [comment]
Recommend you look at https://www.reactiveui.net/docs/handbook/data-binding/ and https://www.reactiveui.net/docs/handbook/data-binding/windows-presentation-foundation

Main things to look out for is using a `IViewFor<TViewModel>`, in WPF there are a number of controls that implement that for you discussed in the second article. The binding engine looks out for this interface to determine the ViewModel type.

**Reply by mGKScoder** [reply]
thanks. I will try my best I don't want to lose my job. But i have quick question. In mvvm design pattern we shouldn't write anything to code behind i mean we shouldn't write anything to behind view. Will this cause problems with mvvm logic ?

**Reply by ChrisPulman** [reply]
`Model` is your data source, View is where this data is displayed, `ViewModel` is the functionality and methods to provide properties for your view.
In the `View` code behind you will create all of your bindings as shown in the link provided by Glenn. This give a data flow of `Model` > `ViewModel` > `View`. The reverse of this can be achieved by Commands also bound in the code behind of the View to a `ReactiveCommand` created in the `ViewModel` a `Button` inherents from a `ICommand` so is one of the controls that could be used for this. By reading the documentation you should be able to obtain an understanding of how ReactiveUI works with MVVM there's also many samples demonstrating some of this functionality.

**Reply by mGKScoder** [reply]
I guess i have an idea now. thank you for all your advice. If I finish my project I will share it here

**Comment by glennawatson** [comment]
MVVM dictates that you don't have logic other than what the view does in it's code behind, nothing about not using the code behind.

Binding's are purely a View concern so it doesn't break any MVVM mandate.

**Comment by glennawatson** [comment]
Here's a sample you can build upon for WPF. https://github.com/reactiveui/ReactiveUI.Samples/tree/main/wpf/getting-started/ReactiveDemo

**Reply by mGKScoder** [reply]
thank you so much
